Core/OfficeDevPnP.Core/Framework/Provisioning/ObjectHandlers/ObjectPropertyBagEntry.cs

```diff
@@ -36,7 +36,7 @@ public override TokenParser ProvisionObjects(Web web, ProvisioningTemplate templ
 
                 // To handle situations where the propertybag is not updated fully when applying a theme, 
                 // we need to create a new context and use that one. Reloading the propertybag does not solve this.
-                var newContext = web.Context.Clone(web.Context.Url);
+                var newContext = web.Context.Clone(web.Url);
```

Comment:
It is not a 100% given that the Url property has been loaded. Better would be to make sure it's available by means of

```csharp
var webUrl = web.EnsureProperty(w => w.Url);
var newContext = web.Context.Clone(webUrl);
```

This will load the Url property only if has not been loaded yet.